examples/sat/schoening.py

```diff
@@ -1,14 +1,31 @@
 """This module provides the Schöning example for solving SAT instances."""
 from typing import Optional
 import numpy as np
-import itertools
 from sat import SatInstance, Assignment
 from qubrabench.stats import QueryStats
-from qubrabench.algorithms.search import search
+from qubrabench.algorithms.search import search_custom, SearchDomain
 
 __all__ = ["schoening_solve"]
 
 
+class SchoeningDomain(SearchDomain):
```

Comment:
this should inherit `SearchDomain[E]` right? with the appropriate element type `E`.